code/client/clrcore/External/Weapon.cs

```diff
@@ -251,6 +333,39 @@ public bool CanUseOnParachute
 			}
 		}
 
+		public bool IsMk2
+		{
+			get
+			{
+				switch (Hash)
+				{
+					case WeaponHash.PistolMk2:
+					case WeaponHash.AssaultRifleMk2:
+					case WeaponHash.CarbineRifleMk2:
+					case WeaponHash.CombatMGMk2:
+					case WeaponHash.HeavySniperMk2:
+					case WeaponHash.SMGMk2:
+						return true;
+					default:
+						return false;
+				}
+			}
+		}
+
+		public void SetLivery(WeaponLivery liveryID, WeaponLiveryColor colorID)
+		{
+			if (IsMk2)
+			{
+				WeaponComponent comp = Components.GetMk2CamoComponent((int)liveryID);
+				comp.Active = true;
+				Function.Call((Hash)0x9FE5633880ECD8ED, Game.PlayerPed.Handle, Hash, comp.ComponentHash, (int) colorID);
```

Comment:
I cannot find it in the natives reference, and as far as I know it has not been given a name yet.
Source of info is http://gtaforums.com/topic/890508-gta-v-gunrunning-weapon-livery-colors/?p=1069683952.